examples/inject/src/main/java/io/helidon/examples/inject/CovarianceExample.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.examples.inject;
+
+import java.util.List;
+
+import io.helidon.service.registry.Service;
+import io.helidon.service.registry.ServiceRegistryManager;
+
+/**
+ * An example that demonstrates covariant lookups.
+ */
+class CovarianceExample {
+
+    private CovarianceExample() {
+    }
+
+    sealed interface Pet permits Cat, Dog {
+        default String name() {
+            return getClass().getSimpleName();
+        }
+    }
+
+    sealed interface Cat extends Pet permits Siamese, Bengal {
+    }
+
+    sealed interface Dog extends Pet permits Boxer, Husky {
+    }
+
+    @Service.Singleton
+    record Siamese() implements Cat {
+    }
+
+    @Service.Singleton
+    record Bengal() implements Cat {
+    }
+
+    @Service.Singleton
+    record Boxer() implements Dog {
+    }
+
+    @Service.Singleton
+    record Husky() implements Dog {
+    }
+
+    @Service.Singleton
+    record Shelter(List<Pet> all, List<Cat> cats, List<Dog> dogs) {
+    }
+
+    public static void main(String[] args) {
```

Comment:
In all our classes (and in our code formatting setup) we have methods first, classes later.

Reply:
It messes the reading, because the main class code comes before.
Can we make an exception or shall I go ahead and move the main methods in all the files ?

Reply:
I would prefer this fixed, as this will get re-arranged when I run code format, and that is the reason why we have code format specified in the project...

Reply:
Specified where ? The checkstyle check is passing.

Reply:
I have moved all the method methods at the top.

Reply:
Specified here: https://github.com/helidon-io/helidon/blob/main/etc/codestyle/idea-code-style.xml

Reply:
Unless we can run this file in a GitHub action, we shouldn't require any changes based on this file since we can't enforce anything.